AK/Time.h

```diff
@@ -354,6 +354,125 @@ class Duration {
     u32 m_nanoseconds { 0 }; // Always less than 1'000'000'000
 };
 
+namespace Detail {
+
+// Common base class for all naive time types. This class and its children are unaware of timezones.
+struct NaiveTime {
```

Comment:
Instead of calling things "naive" (which can be both a temporal and relative assessment), maybe call things like they are. Instead of having a comment explain what it is, why not call it something like `TimezoneUnawareTime`?

Reply:
"naive" time is a term I have seen consistently being used as a specific classifier in Java and Python time libraries, the former of which is usually considered an excellent time handling library. I was therefore borrowing terminology I was familiar with in this context.

Reply:
I'm not aware of that terminology being used in Java, but in Python it seems to be pitted against `aware` as a term. I like borrowing good concepts from other languages, but I don't really see a case why we shouldn't make it more explicit instead of the pretty vague "naive". I.e.: if other developers (such as myself) have not encountered this terminology before, it's not immediately clear what the class does or represents. That to me is an indication that we can figure out a better name.

Reply:
I have added more clarification to the comment, I hope it's okay this way.

Reply:
It already was clear to me from the comment what `NaiveTime` is representing, that was not the issue; why have a comment if you can have a _name_ to explain what is does? That name will pop up in code, autocompletion, etc. and does not require the reader to find the definition and read peripheral comments.

If anything, we need fewer comments, not more :^)